src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,895 @@
+		// the enter or right arrow key.
+
+		protected override void OnKeyDown(KeyRoutedEventArgs args)
+		{
+			base.OnKeyDown(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnKeyDown.", this));
+#endif // MFSI_DEBUG
+
+			bool handled = false;
+			handled = args.Handled;
+			bool shouldHandleEvent = false;
+
+			if (!handled)
+			{
+				MenuFlyoutPresenter spParentPresenter = GetParentMenuFlyoutPresenter();
+
+				if (spParentPresenter != null)
+				{
+					var key = VirtualKey.None;
+
+					key = args.Key;
+
+					// Navigate each item with the arrow down or up key
+					if (key == VirtualKey.Down || key == VirtualKey.Up)
+					{
+						spParentPresenter.HandleUpOrDownKey(key == VirtualKey.Down);
+						UpdateVisualState();
+
+						// If we handle the event here, it won't get handled in m_menuHelper.OnKeyDown,
+						// so we'll do that afterwards.
+						shouldHandleEvent = true;
+					}
+				}
+			}
+
+			m_menuHelper.OnKeyDown(args);
+			args.Handled = !!shouldHandleEvent;
+		}
+
+
+		protected override void OnKeyUp(KeyRoutedEventArgs args)
+		{
+			base.OnKeyUp(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnKeyUp.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnKeyUp(args);
+		}
+
+		// Ensure the creating the popup and menu presenter to show the 
+		void EnsurePopupAndPresenter()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: EnsurePopupAndPresenter.", this));
+#endif // MFSI_DEBUG
+
+			if (m_tpPopup == null)
+			{
+				MenuFlyoutPresenter spParentMenuFlyoutPresenter = null;
+				Control spPresenter;
+				UIElement spPresenterAsUI;
+				FrameworkElement spPresenterAsFE;
+				Popup spPopup;
+
+				spPopup = new Popup();
+				spPopup.IsSubMenu = true;
+				spPopup.IsLightDismissEnabled = false;
+
+#if false // UNO TODO Windowed Popup is not available
+				spParentMenuFlyoutPresenter = GetParentMenuFlyoutPresenter();
+				if (spParentMenuFlyoutPresenter != null)
+				{
+					spParentMenuFlyout = spParentMenuFlyoutPresenter.GetParentMenuFlyout();
+					// Set the windowed Popup if the MenuFlyout is set the windowed Popup
+					if (spParentMenuFlyout && spParentMenuFlyout.IsWindowedPopup())
+					{
+						ASSERT((CPopup*)(spPopup as Popup.GetHandle()).DoesPlatformSupportWindowedPopup(DXamlCore.GetCurrent().GetHandle()));
+
+						((CPopup*)(spPopup as Popup.GetHandle()).SetIsWindowed());
+
+						// Ensure the sub menu is the windowed Popup
+						ASSERT((CPopup*)(spPopup as Popup.GetHandle()).IsWindowed());
+
+						xaml.IXamlRoot xamlRoot = XamlRoot.GetForElementStatic(spParentMenuFlyoutPresenter);
+						if (xamlRoot)
+						{
+							(spPopup as Popup.XamlRoot = xamlRoot);
+						}
+					}
+				}
+#endif
+
+				spPresenter = CreateSubPresenter();
+				spPresenterAsUI = spPresenter;
+
+				if (spParentMenuFlyoutPresenter != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)